src/dstack/_internal/gateway/deps.py

```diff
@@ -0,0 +1,28 @@
+from abc import ABC, abstractmethod
+from typing import AsyncGenerator
+
+from fastapi import Depends, Request
+from typing_extensions import Annotated
+
+from dstack._internal.gateway.repos.base import BaseGatewayRepo
+
+
+class BaseGatewayDependencyInjector(ABC):
```

Comment:
This Injector approach requires explanation, i.e. comment that we use it so that gateway routes can depend on different injectors when running on a server and on a remote gateway.